### ISSUE TITLE
Config 4:  Test SSP IMEX with limiter, with the most stable configuration (2 or 3 Newton iterations), using `:zalesak` vertical upwinding for tracers

### ISSUE THREAD

**Issue by valeriabarra** [CliMA/ClimaAtmos.jl#1513]
Part of #1310

**Comment by valeriabarra** [comment]
Reopening as I had accidentally linked the wrong issue in the PR

**Comment by valeriabarra** [comment]
In the investigation of the `zalesak` upwinding, I tested separately the first-order upwinding alone and the third-order upwinding alone (since `zalesak` is a combination of the two). 
The first-order upwinding alone and the third-order upwinding alone pass the regular CI (see PRs #1532 #1534 and #1541) and are currently tested in the longrun one. 

Zalesak is currently tested in both the regular and the longrun pipelines.

**Comment by valeriabarra** [comment]
This configuration is unstable also in the regular (coarse) pipeline set up. 

This is the longest I could get the moist baroclinic wave to run: 3 days, with the following CLI options:
```
julia --color=yes --project=examples examples/hybrid/driver.jl --initial_condition MoistBaroclinicWave --moist equil --precip_model 0M --job_id sphere_zalesak_upwind_ssp_baroclinic_wave_rhoe_equilmoist --regression_test true --dt 150secs --dt_save_to_disk 5days --ode_algo SSP333 --max_newton_iters 3 --tracer_upwinding zalesak
``` 

At day 3, right before the crash, this is how the `qt` looks like, at different vertical levels (10 level in total): 
- `qt`, day 3, level 7:
![rho_q_tot_day3_level7](https://user-images.githubusercontent.com/39932030/230256196-ad01fd22-b36c-4808-9b33-6d0a0d7aea11.png)

- `qt`, day 3, level 8:
![rho_q_tot_day3_level8](https://user-images.githubusercontent.com/39932030/230256257-b9a98b85-6dc9-4041-8455-8560177110ff.png)

- `qt`, day 3, level 9:
![rho_q_tot_day3_level9](https://user-images.githubusercontent.com/39932030/230256282-a5b5b273-f983-4425-938a-2304ac41469e.png)

- `qt`, day 3, level 10:
![rho_q_tot_day3_level10](https://user-images.githubusercontent.com/39932030/230256300-e7ab9860-c425-4eb6-b5df-5a6fb527ca5a.png)


and here are the extrema for `qt`:

```
println(ρq_tot |> parent |> extrema)
(-5.692269f-10, 0.017981058f0)
```

I tried several time steps and Newton iterations but nothing helped. 

I also tried running explicit only (to rule out any potential problem with the approximated Jacobian implementation), but it didn't help either. 

I think we can close this issue and investigate this separately.
